p2p/router.go

```diff
@@ -544,3 +522,16 @@ func (r *Router) OnStop() {
 		<-q.closed()
 	}
 }
+
+// stopCtx returns a context that is cancelled when the router stops.
+func (r *Router) stopCtx() context.Context {
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		select {
+		case <-r.stopCh:
+			cancel()
+		case <-ctx.Done():
```

Comment:
not sure if this makes sense. ctx.Done() is closed when `cancel()` is called, but since cancel() is called above upon r.stopCh, this line will never be executed.

Reply:
That's true. I figured I'd cover the case in case it was cancelled for whatever other (external) reason, but I suppose that's not currently possible.